modules/postgresql/main.tf

```diff
@@ -173,6 +174,45 @@ resource "google_sql_user" "additional_users" {
   depends_on = [null_resource.module_depends_on, google_sql_database_instance.default]
 }
 
+resource "google_sql_user" "iam_account" {
+  for_each = local.iam_users
+  project  = var.project_id
+  name = each.value.is_account_sa ? (
+    trimsuffix(google_service_account.sa[each.value.name].email, ".gserviceaccount.com")
+    ) : (
+    each.value.name
+  )
+  instance = google_sql_database_instance.default.name
+  type     = each.value.is_account_sa ? "CLOUD_IAM_SERVICE_ACCOUNT" : "CLOUD_IAM_USER"
+  depends_on = [
+    null_resource.module_depends_on,
+    google_sql_database_instance.default,
+    google_service_account.sa,
+  ]
+}
+
+resource "google_service_account" "sa" {
+  for_each = {
+    for i in local.iam_users : i.name => i
+    if i.is_account_sa
+  }
+
+  project      = var.project_id
+  account_id   = each.value.name
+  display_name = each.value.description
+}
```

Comment:
This would create a new SA. IIUC the ask from https://github.com/terraform-google-modules/terraform-google-sql-db/issues/201 is to allow users to pass in a SA email like `sa-name@project-id.iam.gserviceaccount.com ` and have some logic within the module to convert that into ` sa-name@project-id.iam`

Reply:
Yeah that's how I started implementing it but then I thought it would be much easier if the module allowed for creation of the SA as well. This way all the user has to do is pass in the sa-name and the module creates the sa (if not already present) and does the trimming logic necessary to add it to postgresql as a user.

Reply:
And I believe they are using the `google_service_account` resource in their module to provision an SA. I think the whole creation & trimming of the SA should be abstracted from the user and done by the module as long as they pass in a sa-name.

Reply:
We should not create service accounts in this module.

The general principle is that service accounts should be created where they are *used* (ie. near the resource which will use it as an "identity") not where they will be granted permission.

Reply:
That makes sense to me now. I'll change this accordingly.